### DISCUSSION TITLE
Teleport with embedded, IoT devices and busybox

### DISCUSSION THREAD

**Discussion by dperigault** [gravitational/teleport#18246] [Q&A] [answered]
Hello,

Im thinking to use Teleport to control the ssh of the  my infra but I have couple of devices I can SSH on it but I cannot really install any extra package (small IoT devices, switch,router,servers BMC, etc etc)

Im not exactly sure how I can achieve that and if that's possible with Teleport?

Thanks!

**Comment by webvictim** [comment, marked as answer]
If you can edit the `/etc/ssh/sshd_config` file on your devices then you can probably tell them to trust Teleport-issued user certificates for authentication and connect in an agentless way. See our OpenSSH guide for some more details: https://goteleport.com/docs/server-access/guides/openssh/#step-14-configure-sshd-to-trust-the-teleport-ca

**Comment by dperigault** [comment]
I just tested it (thanks!) but Teleport cluster will not be aware of the host I "register" that way. Is that possible to change this?I believe this is also why the hosts I added like that are not visible with a `tsh ls`

**Reply by webvictim** [reply]
This is correct - not running the Teleport agent service comes at a cost, which is that nodes can't register themselves.

It is technically possible to register SSH services directly in the Teleport backend via API but we haven't written any docs on how to do this yet.

**Comment by marvin-enthus** [comment]
Hi. I'm also jumping into this discussion as I'm also evaluating whether Teleport can be used to manage network infrastructure. All of these network devices are locked, so we have no chance to modify the SSH configuration to allow authentication via a CA as required for this concept: https://goteleport.com/docs/server-access/guides/openssh/#step-14-configure-sshd-to-trust-the-teleport-ca 
A few products allow the configuration of individual 'ssh certificates', but this is only allowed by very few products. But most products support username/password authentication with validation by a radius server. Usually you also can control user access via special radius attributes. So in kombination with a radius server we could generate 'single use' passwords for authenticate this specific session. Are there any plans to bring a 'username/pw' SSH auth mode for network devices? (for the first shot just username/pw without radius)

**Reply by webvictim** [reply]
Teleport as a product is all about getting away from usernames/passwords rather than going back towards them, so unfortunately I doubt that's something we're ever likely to do. The RADIUS integration for a one-time password (like a token) is definitely an interesting idea that I've wanted to explore in the past, but it isn't something that's currently on our roadmap.
